lxd/instance/drivers/driver_qemu.go

```diff
@@ -3070,20 +3070,15 @@ func (d *qemu) addNetDevConfig(cpuCount int, busName string, qemuDev map[string]
 		}
 	}
 
-	// If the devices' configured name has no value, fallback to the device key/name.
-	if name == "" {
-		name = devName
-	}
-
 	if shared.IsTrue(d.expandedConfig["agent.rename_interfaces"]) {
-		err := d.writeNICDevConfig(mtu, name, devHwaddr)
+		err := d.writeNICDevConfig(mtu, devName, name, devHwaddr)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("Failed writing NIC config for device %q: %w", devName, err)
 		}
 	}
 
 	var qemuNetDev map[string]interface{}
-	qemuDev["id"] = fmt.Sprintf("%s%s", qemuDeviceIDPrefix, devName)
+	qemuDev["id"] = fmt.Sprintf("%s%s", qemuDeviceIDPrefix, filesystem.PathNameEncode(devName))
```

Comment:
> 2\. The QEMU device config doesn't like `/` in the device IDs.

I thought we had disallowed `/` from device names already to make them compatible as file names?